pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -242,15 +222,32 @@ func (s libInfoSource) isFromLanguageDetection() bool {
 	return s == libInfoSourceSingleStepLangaugeDetection
 }
 
-func (s libInfoSource) mutatePod(pod *corev1.Pod) error {
-	_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
-		Name:  instrumentationInstallTypeEnvVarName,
-		Value: s.injectionType(),
-	})
+func (s libInfoSource) instrumentationInstallTime() string {
+	instrumentationInstallTime := os.Getenv(instrumentationInstallTimeEnvVarName)
+	if instrumentationInstallTime == "" {
+		instrumentationInstallTime = common.ClusterAgentStartTime
+	}
 
-	injectApmTelemetryConfig(pod)
+	return instrumentationInstallTime
+}
 
-	return nil
+func (s libInfoSource) podMutator(filter containerPredicate) podMutator {
```

Comment:
Should document what the `podMutator` mutates. What are it's responsibilities?

Reply:
Updated this to produce a `containerMutator` so we can use `m.core.mutatePodContainers` which applies the filter _for us_.